Platform/OpenNetworkBoot/BmBoot.c

```diff
@@ -12,2653 +12,535 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
-  @retval FALSE The boot option is manually added.
-**/
-BOOLEAN
-BmIsAutoCreateBootOption (
-  EFI_BOOT_MANAGER_LOAD_OPTION  *BootOption
-  )
-{
-  if ((BootOption->OptionalDataSize == sizeof (EFI_GUID)) &&
-      CompareGuid ((EFI_GUID *)BootOption->OptionalData, &mBmAutoCreateBootOptionGuid)
-      )
-  {
-    return TRUE;
-  } else {
-    return FALSE;
-  }
-}
-
 /**
-  Find the boot option in the NV storage and return the option number.
-
-  @param OptionToFind  Boot option to be checked.
+  Expand the media device path which points to a BlockIo or SimpleFileSystem instance
+  by appending EFI_REMOVABLE_MEDIA_FILE_NAME.
 
-  @return   The option number of the found boot option.
+  @param DevicePath  The media device path pointing to a BlockIo or SimpleFileSystem instance.
+  @param FullPath    The full path returned by the routine in last call.
+                     Set to NULL in first call.
 
+  @return The next possible full path pointing to the load option.
+          Caller is responsible to free the memory.
 **/
-UINTN
-BmFindBootOptionInVariable (
-  IN  EFI_BOOT_MANAGER_LOAD_OPTION  *OptionToFind
+EFI_DEVICE_PATH_PROTOCOL *
+BmExpandMediaDevicePath (
+  IN  EFI_DEVICE_PATH_PROTOCOL  *DevicePath,
+  IN  EFI_DEVICE_PATH_PROTOCOL  *FullPath
   )
 {
-  EFI_STATUS                    Status;
-  EFI_BOOT_MANAGER_LOAD_OPTION  BootOption;
-  UINTN                         OptionNumber;
-  CHAR16                        OptionName[BM_OPTION_NAME_LEN];
-  EFI_BOOT_MANAGER_LOAD_OPTION  *BootOptions;
-  UINTN                         BootOptionCount;
-  UINTN                         Index;
-
-  OptionNumber = LoadOptionNumberUnassigned;
-
-  //
-  // Try to match the variable exactly if the option number is assigned
-  //
-  if (OptionToFind->OptionNumber != LoadOptionNumberUnassigned) {
-    UnicodeSPrint (
-      OptionName,
-      sizeof (OptionName),
-      L"%s%04x",
-      mBmLoadOptionName[OptionToFind->OptionType],
-      OptionToFind->OptionNumber
-      );
-    Status = EfiBootManagerVariableToLoadOption (OptionName, &BootOption);
-
-    if (!EFI_ERROR (Status)) {
-      ASSERT (OptionToFind->OptionNumber == BootOption.OptionNumber);
-      if ((OptionToFind->Attributes == BootOption.Attributes) &&
-          (StrCmp (OptionToFind->Description, BootOption.Description) == 0) &&
-          (CompareMem (OptionToFind->FilePath, BootOption.FilePath, GetDevicePathSize (OptionToFind->FilePath)) == 0) &&
-          (OptionToFind->OptionalDataSize == BootOption.OptionalDataSize) &&
-          (CompareMem (OptionToFind->OptionalData, BootOption.OptionalData, OptionToFind->OptionalDataSize) == 0)
-          )
-      {
-        OptionNumber = OptionToFind->OptionNumber;
-      }
-
-      EfiBootManagerFreeLoadOption (&BootOption);
-    }
-  }
+  EFI_STATUS                Status;
+  EFI_HANDLE                Handle;
+  EFI_BLOCK_IO_PROTOCOL     *BlockIo;
+  VOID                      *Buffer;
+  EFI_DEVICE_PATH_PROTOCOL  *TempDevicePath;
```

Comment:
The diff for this file (here and onwards), at this second commit, is very misleading; this commit is _only_ simple deletes of lines, as can be seen e.g. with `git diff --minimal network-boot~2 network-boot~` at the command line.